seed/data_importer/match.py

```diff
@@ -548,86 +559,91 @@ def states_to_views(unmatched_state_ids, org, access_level_instance, cycle, Stat
     # -States with all matching criteria being None. These aren't eligible for matching.
-    empty_matching_criteria = empty_criteria_filter(StateClass, column_names)
-    promote_states = StateClass.objects.filter(pk__in=unmatched_state_ids, **empty_matching_criteria).exclude(
-        pk__in=Subquery(duplicate_states.values("id"))
+    empty_matching_criteria = empty_criteria_filter(StateClass, matching_columns)
+    promote_state_ids = set(
+        StateClass.objects.filter(pk__in=unmatched_state_ids.difference(duplicate_state_ids), **empty_matching_criteria).values_list(
+            "id", flat=True
+        )
     )
 
     # Identify and filter out -States that have been "handled".
-    handled_states = promote_states | duplicate_states
-    unmatched_states = StateClass.objects.filter(pk__in=unmatched_state_ids).exclude(pk__in=Subquery(handled_states.values("id")))
+    handled_state_ids = promote_state_ids.union(duplicate_state_ids)
+    unmatched_states = StateClass.objects.filter(pk__in=unmatched_state_ids.difference(handled_state_ids))
 
     # For the remaining -States, search for a match within the -States that are attached to -Views.
     # If one match is found, pass that along.
     # If multiple matches are found, merge them together, pass along the resulting record.
-    # Otherwise, add current -State to be promoted as is.
+    # Otherwise, add current -State to be promoted as-is.
     merged_between_existing_count = 0
-    merge_state_pairs = []
+    merge_state_pairs: list[Union[tuple[PropertyState, PropertyState], Union[TaxLotState, TaxLotState]]] = []
     batch_size = math.ceil(len(unmatched_states) / 100)
 
     for idx, state in enumerate(unmatched_states):
-        matching_criteria = matching_filter_criteria(state, column_names)
+        matching_criteria = matching_filter_criteria(state, matching_columns)
 
         # compare UBIDs via jaccard index instead of a direct match
         check_jaccard = bool(matching_criteria.get("ubid"))
-        if check_jaccard:
-            # Only pop the UBID value if it's populated
-            # If it's `None` then we still want to match against existing records with empty UBIDs
-            ubid = matching_criteria.pop("ubid")
 
-        existing_state_matches = StateClass.objects.filter(
-            pk__in=Subquery(existing_cycle_views.values("state_id")),
-            **matching_criteria,
-        )
+        existing_state_matches = match_lookup.get(tuple(getattr(state, c) for c in tuple_values), [])
 
         if check_jaccard:
             existing_state_matches = [
-                state for state in existing_state_matches if state.ubid and check_jaccard_match(ubid, state.ubid, org.ubid_threshold)
+                state
+                for state in existing_state_matches
+                if state.get("ubid") and check_jaccard_match(matching_criteria.get("ubid"), state.get("ubid"), org.ubid_threshold)
```

Comment:
idk how this state and the one declared on L612, arent clashing, but could change this variable name please?

Reply:
I'll change it, but variables created within a list comprehension are scoped only within the comprehension itself, so the original `state` variable is unaffected